contracts/interchain-token/tests/test.rs

```diff
@@ -4,55 +4,515 @@ extern crate std;
+        symbol: "symbol".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn token_symbol_is_empty() {
+    let env = Env::default();
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: 1,
+        name: "name".into_val(&env),
+        symbol: "".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn token_id_is_empty() {
+    let env = Env::default();
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: 1,
+        name: "name".into_val(&env),
+        symbol: "".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
         ),
-        (),
     );
+}
+
+#[test]
+#[should_panic(expected = "insufficient balance")]
+fn transfer_insufficient_balance() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let token = create_token(&env, &owner, &minter);
+
+    token.mint(&minter, &user1, &1000);
+    assert_eq!(token.balance(&user1), 1000);
 
-    assert_eq!(client.owner(), new_owner);
+    token.transfer(&user1, &user2, &1001);
 }
 
 #[test]
-fn transfer_ownership_unauthorized() {
-    let (env, client) = setup_env();
-    let new_owner = Address::generate(&env);
-    assert_invoke_auth_err!(new_owner, client.try_transfer_ownership(&new_owner));
+#[should_panic(expected = "insufficient allowance")]
+fn transfer_from_insufficient_allowance() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let user3 = Address::generate(&env);
+    let token = create_token(&env, &owner, &minter);
+
+    token.mint(&minter, &user1, &1000);
+    assert_eq!(token.balance(&user1), 1000);
```

Comment:
since mint isn't being tested here, an assert for the mint itself isn't needed

Reply:
make sense. updated